### PR TITLE
[X86] Support immediate folding for CCMP/CTEST

### PR DESCRIPTION
**Pull request by FreddyLeaf**
E.g.
```
%0:gr32 = MOV32ri 81
CTEST32rr %0, %1, 2, 10, implicit-def $eflags, implicit $eflags
=>
CTEST32ri %1, 81, 2, 10, implicit-def $eflags, implicit $eflags
```

### CODE REVIEW

**Comment by KanRobert** [comment]
Could you elaborate what immediate folding does for CCMP/CTEST?

**Comment by FreddyLeaf** [comment]
> Could you elaborate what immediate folding does for CCMP/CTEST?

Done in commit message.

File: llvm/test/CodeGen/X86/apx/foldimmediate.mir [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    CTEST32rr %0, %1, 2, 10, implicit-def $eflags, implicit $eflags
+    NOOP implicit $eflags
+
+    CCMP32rr %1, %0, 2, 10, implicit-def $eflags, implicit $eflags
+    NOOP implicit $eflags
+
+    %16 = ADC32rr %0, %1, implicit-def $eflags, implicit $eflags
+    NOOP implicit %16
+
+    %18 = SBB32rr %1, %0, implicit-def $eflags, implicit $eflags
```
Comment (KanRobert) [inline llvm/test/CodeGen/X86/apx/foldimmediate.mir:107]: Reduce the test. We don't need to sth like SBB, AND

Comment (FreddyLeaf) [reply llvm/test/CodeGen/X86/apx/foldimmediate.mir:107]: This is copied from llvm/test/CodeGen/X86/foldimmediate.mir

Comment (FreddyLeaf) [reply llvm/test/CodeGen/X86/apx/foldimmediate.mir:107]: 19168512547a5577077a88b66dd03d92a5b28dc7 can directly show the patch impact, though there are some mistakes fixed by following commit.

Comment (FreddyLeaf) [reply llvm/test/CodeGen/X86/apx/foldimmediate.mir:107]: Never mind. I'll reduce.

Comment (FreddyLeaf) [reply llvm/test/CodeGen/X86/apx/foldimmediate.mir]: [4b18e06](https://github.com/llvm/llvm-project/pull/86616/commits/4b18e06671c5f755e48ca0f6c53b3814c756c118)

**Review by KanRobert — APPROVED** [review]
LGTM

**Pull request merged**